### ISSUE TITLE
CppParser.ToggleSourceHeader is a bit slow

### ISSUE THREAD

**Issue by hsandt** [slynch8/10x#2614]
CppParser.ToggleSourceHeader (default: Alt+O) takes 0.5s to 1s to switch across files. This is not *very slow*, but it's not super fast either for two files that are either siblings, or only apart by 1 level of folders (think Public/MyClass.h vs Private/MyClass.cpp in Unreal projects).

I noticed the same issue with Visual Studio (without VAX I think), while other code editors like SublimeText, who do less scanning to make sure that files are really semantically linked, and more filename check, are faster.

Since the current implementation seems to find matching files by name _anyway_ and may mess up finding wrong, unrelated files (see issues such as https://github.com/slynch8/10x/issues/2159), it may as well be fast. I suppose that implementing the latter issue, going to the closest file, will naturally also fix mine, since it would then be very fast (and only slow when the related file is far away in the hierarchy - in which case we'll need relationship caching).

We can always implement true semantic relationship check later, in which case I would accept to have a slower command. For instance, we could have a generic, super fast, Sublime-Text-style GoToRelatedFile that only checks filenames with different extensions (and possibly a custom prefix/suffix such as `test_` or `_test`), and the more traditional, semantic, Visual-Studio-style CppParser.ToggleSourceHeader to find the header/source really matching the other file.

**Comment by slynch8** [comment]
You're right, it's not doing anything that clever at the moment, so it should be instant. I'll take a look.

**Comment by MohitSethi99** [comment]
Tried it out in the early build and it seems to be picking up wrong file.

I was trying to toggle between these two files.
```
Arc/src/Arc/Core/JobSystem.cpp
Arc/src/Arc/Core/JobSystem.h
```

But it was toggling to these ones.
```
Arc/vendor/JoltPhysics/JoltPhysics/Jolt/Core/JobSystem.h
Arc/vendor/JoltPhysics/JoltPhysics/Jolt/Core/JobSystem.inl
```

Here's the test project: https://github.com/MohitSethi99/ArcGameEngine
branch: dx12

**Comment by slynch8** [comment]
fixed (1.0.168)
